core/src/test/java/org/apache/iceberg/util/TestParallelIterable.java

```diff
@@ -60,9 +62,12 @@ public CloseableIterator<Integer> iterator() {
 
     assertThat(iterator.hasNext()).isTrue();
     assertThat(iterator.next()).isNotNull();
-    assertThat(queue).isNotEmpty();
-
+    Awaitility.await("Queue is populated")
+        .atMost(5, TimeUnit.SECONDS)
+        .untilAsserted(() -> assertThat(queue).isNotEmpty());
```

Comment:
Actually the same may apply for the very first element offered to the queue, will fix that